pkg/controller/controller_test.go

```
@@ -0,0 +1,357 @@
+		{3, "4s"},
+		{4, "8s"},
+		{5, "10s"}, // capped at maxRetryBackoff
+	}
+	for _, tt := range tests {
+		got := calculateBackoff(tt.attempt)
+		if got.String() != tt.want {
+			t.Errorf("calculateBackoff(%d) = %v, want %v", tt.attempt, got, tt.want)
+		}
+	}
+}
+
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofmt)